api/desecapi/tests/test_rrsets.py

```diff
@@ -1636,3 +1638,131 @@ def test_create_my_rr_sets_ip_block(self):
         )
         self.assertStatus(response, status.HTTP_400_BAD_REQUEST)
         self.assertIn("IP address 3.2.2.5 not allowed.", str(response.data))
+
+    def test_create_ns_rrset(self):
+        for subname in ["", "sub"]:
+            data = dict(self.ns_data, subname=subname)
+            with self.assertNoRequestsBut():
+                self.assertBadRequest(
+                    self.client.post_rr_set(
+                        domain_name=self.my_empty_domain.name, **data
+                    ),
+                    "Cannot modify NS records for this domain.",
+                    ("type", 0),
+                )
+
+    def test_update_ns_rrset(self):
+        for subname in ["", "sub"]:
+            data = dict(self.ns_data, subname=subname)
+            self.create_rr_set(
+                self.my_domain,
+                settings.DEFAULT_NS,
+                subname=subname,
+                type="NS",
+                ttl=3600,
+            )
+            for method in (self.client.patch_rr_set, self.client.put_rr_set):
+                with self.assertNoRequestsBut():
+                    self.assertBadRequest(
+                        method(self.my_domain.name, subname, "NS", data),
+                        "Cannot modify NS records for this domain.",
+                        ("type", 0),
+                    )
+
+    def test_delete_ns_rrset_apex(self):
+        data = dict(self.ns_data, records=[], subname="")
+        self.create_rr_set(
+            self.my_domain, settings.DEFAULT_NS, subname="", type="NS", ttl=3600
+        )
+        for method in (self.client.patch_rr_set, self.client.put_rr_set):
+            with self.assertNoRequestsBut():
+                self.assertBadRequest(
+                    method(self.my_domain.name, "", "NS", data),
+                    "Cannot modify NS records for this domain.",
+                    ("type", 0),
+                )
+        with self.assertNoRequestsBut():
+            response = self.client.delete_rr_set(self.my_domain.name, "", "NS")
+            self.assertStatus(response, status.HTTP_400_BAD_REQUEST)
+
+    def test_delete_ns_rrset_nonapex(self):
+        data = dict(self.ns_data, subname="sub", records=[])
+        for method in (self.client.patch_rr_set, self.client.put_rr_set):
+            self.create_rr_set(
+                self.my_domain, settings.DEFAULT_NS, subname="sub", type="NS", ttl=3600
+            )
+            with self.assertRequests(
+                self.requests_desec_rr_sets_update(name=self.my_domain.name)
+            ):
+                response = method(self.my_domain.name, "sub", "NS", data)
```

Comment:
subname arg seems redundant - once as argument to `method` and once in `data`.

Reply:
No, PUT requires sending the full object